ubuntudesign/docker-devrun
# Support building of sass files

For the time being, `./run` will build sass by simply calling a `watch` command which I expect to be defined in the `package.json` local to the project (see `resources/package.json` in this branch).

Will and I have had a discussion around potential solutions. The eventual solution will probably do a better job of encapsulating NPM tasks, to reduce duplication across projects even further.

So what's done here:

-  Make use of the new [node image](https://github.com/ubuntudesign/docker-node) to support running of `node`, `npm` and `gulp` commands in a context where any dependencies defined in `package.json` will be automatically installed and updated as needed.
- Give the `server` command a `--watch` option, which will run `npm run-script watch`, within the aforementioned `node` image
- Make `server --watch` the default command which will be run on simply calling `./run`

QA
--

Test out this new image in [my `add-devrun` branch of `django-blueprint`](https://github.com/nottrobin/django-blueprint/tree/add-devrun/):

``` bash
git clone -b add-devrun git@github.com:nottrobin/django-blueprint django-blueprint.add-devrun
cd django-blueprint.add-devrun
./run
```

Give it a while to install dependencies - it's ready when you see:

```
watch_1   | Wrote CSS to /app/static/css/global.css
...
webapp_1     | Successfully installed Django-1.8.5
```

Now check the site runs at <http://localhost:8090>.

Kill the server (`<ctrl>+c` *once*), and once it's died start it up again with `./run` - it should be much quicker this time.

Check <http://localhost:8090> again.

Change `static/sass/global.scss` and check that CSS updates promptly and your changes show on the site.
